src/libexpr/tests/coro-gc.cc

```diff
@@ -0,0 +1,147 @@
+#include <gtest/gtest.h>
+#if HAVE_BOEHMGC
+#include <gc/gc.h>
+
+#include "eval.hh"
+#include "serialise.hh"
+
+#endif
+
+
+namespace nix {
+#if HAVE_BOEHMGC
+
+    static void finalizer(void *obj, void *data) {
+        *((bool*)data) = true;
+    }
+
+    static bool* make_witness(volatile void* obj) {
+        /* We can't store the witnesses on the stack,
+           since they might be collected long afterwards */
+        bool* res = (bool*)GC_MALLOC_UNCOLLECTABLE(1);
```

Comment:
```suggestion
        auto res = std::make_shared<bool>(false);
```

I think that would do what we want, and make sure that the GC doesn't mess with it (which it shouldn't, but since is already testing the GC, let's make sure that we properly isolate what we test)

Reply:
Wouldn't work, shared_ptrs are reference counted, but we want `res` to live forever (or at least exceed the lifetime of the thing it's witnessing, which might be forever). Maybe a regular `new` would work.